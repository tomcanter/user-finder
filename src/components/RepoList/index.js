// @flow

import React from 'react';
import {
  StyleSheet,
  css,
} from 'aphrodite/no-important';
import curry from 'lodash/fp/curry';
import map from 'lodash/fp/map';
import Repo from 'components/Repo';
import Loading from 'components/Loading';

type Props = {
  entities: Object,
  ids: Array<number>,
  isPending: boolean,
};

const renderRepoListItem = curry((entities, id) => {
  return (
    <li
      className={css(styles.RepoList_item)}
      key={id}
    >
      <Repo {...entities[id]} />
    </li>
  );
});

export default function RepoList({entities, ids, isPending}: Props) {
  if (isPending) {
    return (
      <Loading />
    );
  }

  return (
    <ul>
      {map(renderRepoListItem(entities), ids)}
    </ul>
  );
}

const styles = StyleSheet.create({
  RepoList_item: {
    borderBottom: '1px solid #ddd',
    marginBottom: 20,
    paddingBottom: 20,

    ':last-child': {
      marginBottom: 0,
      paddingBottom: 0,
      borderBottom: 0,
    },
  },
});
