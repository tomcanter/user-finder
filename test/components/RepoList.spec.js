import React from 'react';
import renderer from 'react-test-renderer';
import MockDate from 'mockdate';
import {
  StyleSheetTestUtils,
} from 'aphrodite';

import RepoList from 'components/RepoList';

describe('Component: RepoList', () => {

  beforeAll(() => {
    MockDate.set(new Date('2017-04-04T09:00:00Z'));
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    MockDate.reset();
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('should render Repo components', () => {
    const entities = {
      123: {
        description: 'My dotfiles, originally forked from Holmans dotfiles.',
        fork: false,
        html_url: 'https://github.com/smgt/dotfiles',
        language: 'Vim script',
        updated_at: '2017-04-04T09:47:48Z',
        name: 'dotfiles',
        stargazers_count: 3,
        forks_count: 0,
      },
      456: {
        description: 'My dotfiles, originally forked from Holmans dotfiles.',
        fork: true,
        html_url: 'https://github.com/smgt/dotfiles',
        language: 'Vim script',
        updated_at: '2017-04-04T09:47:48Z',
        name: 'dotfiles',
        stargazers_count: 3,
        forks_count: 0,
      },
    };
    const component = renderer.create(
      <RepoList
        entities={entities}
        ids={[123, 456]}
        isPending={false}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render a Loading component when isPending', () => {
    const component = renderer.create(
      <RepoList
        entities={{}}
        ids={[]}
        isPending={true}
      />
    );
    expect(component).toMatchSnapshot();
  });

});
