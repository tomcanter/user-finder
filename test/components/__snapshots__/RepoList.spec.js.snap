// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: RepoList should render Repo components 1`] = `
<ul>
  <li
    className="RepoList_item_dgdnyv"
  >
    <div>
      <p
        className="Repo_name_y4kf6o u-flex u-flexAlignItemsBaseline"
      >
        <a
          href="https://github.com/smgt/dotfiles"
        >
          dotfiles
        </a>
      </p>
      <p
        className="Repo_desc_ylvipy"
      >
        My dotfiles, originally forked from Holmans dotfiles.
      </p>
      <div
        className="Repo_footer_de3o52 u-flex u-flexWrap u-flexAlignItemsCenter"
      >
        <p
          className="Repo_footerItem_vaj62s"
        >
          Vim script
        </p>
        <div
          className="Repo_footerItem_vaj62s u-flex u-flexAlignItemsCenter"
        >
          <svg />
          <p>
            3
          </p>
        </div>
        <div
          className="Repo_footerItem_vaj62s u-flex u-flexAlignItemsCenter"
        >
          <svg />
          <p>
            0
          </p>
        </div>
        <p
          className="u-sm-flexExpandLeft"
        >
          updated 
          about 1 hour
           ago
        </p>
      </div>
    </div>
  </li>
  <li
    className="RepoList_item_dgdnyv"
  >
    <div>
      <p
        className="Repo_name_y4kf6o u-flex u-flexAlignItemsBaseline"
      >
        <a
          href="https://github.com/smgt/dotfiles"
        >
          dotfiles
        </a>
        <span
          className="Repo_fork_1im0xn8"
        >
          (fork)
        </span>
      </p>
      <p
        className="Repo_desc_ylvipy"
      >
        My dotfiles, originally forked from Holmans dotfiles.
      </p>
      <div
        className="Repo_footer_de3o52 u-flex u-flexWrap u-flexAlignItemsCenter"
      >
        <p
          className="Repo_footerItem_vaj62s"
        >
          Vim script
        </p>
        <div
          className="Repo_footerItem_vaj62s u-flex u-flexAlignItemsCenter"
        >
          <svg />
          <p>
            3
          </p>
        </div>
        <div
          className="Repo_footerItem_vaj62s u-flex u-flexAlignItemsCenter"
        >
          <svg />
          <p>
            0
          </p>
        </div>
        <p
          className="u-sm-flexExpandLeft"
        >
          updated 
          about 1 hour
           ago
        </p>
      </div>
    </div>
  </li>
</ul>
`;

exports[`Component: RepoList should render a Loading component when isPending 1`] = `
<div
  className="u-flex u-flexJustifyCenter u-flexAlignItemsCenter"
>
  <svg />
</div>
`;
